tests/LinAlg/vectorTest.hpp

```diff
@@ -1527,19 +1537,151 @@ class VectorTests : public TestBase
-  virtual real_type* getLocalData(hiop::hiopVector* x) = 0;
-  virtual int verifyAnswer(hiop::hiopVector* x, real_type answer) = 0;
+  /// Method to set vector _x_ element _i_ to _value_.
+  void setLocalElement(hiop::hiopVector* x, local_ordinal_type i, real_type val)
+  {
+    x->copyFromDev();
+    real_type *xdat = x->local_data_host();
+    xdat[i] = val;
+    x->copyToDev();
+  }
+  /// Returns element _i_ of vector _x_.
+  real_type getLocalElement(const hiop::hiopVector* x, local_ordinal_type i)
+  {
+    x->copyFromDev();
+    return x->local_data_host_const()[i];
+  }
+
+  /// Returns pointer to local vector data
+  local_ordinal_type getLocalSize(const hiop::hiopVector* x)
+  {
+    return static_cast<local_ordinal_type>(x->get_local_size());
+  }
+
+  /// Returns pointer to local vector data
+  virtual real_type* getLocalData(hiop::hiopVector* x)
+  {
+    x->copyFromDev();
+    return x->local_data_host();
+  }
+
+  /// Checks if _local_ vector elements are set to `answer`.
+  int verifyAnswer(hiop::hiopVector* x, real_type answer)
+  {
+    x->copyFromDev();
+    const local_ordinal_type N = getLocalSize(x);
+    auto* xdata = static_cast<const real_type*>(getLocalData(x));
+    
+    int local_fail = 0;
+
+    for(local_ordinal_type i = 0; i < N; ++i)
+    {
+      if(!isEqual(xdata[i], answer))
+      {
+        ++local_fail;
+      }
+    }
+
+    return local_fail;
+  }
+
   virtual int verifyAnswer(
       hiop::hiopVector* x,
-      std::function<real_type(local_ordinal_type)> expect) = 0;
-  virtual bool reduceReturn(int failures, hiop::hiopVector* x) = 0;
-  virtual real_type* createLocalBuffer(local_ordinal_type N, real_type val) = 0;
-  virtual void deleteLocalBuffer(real_type* buffer) = 0;
+      std::function<real_type(local_ordinal_type)> expect)
+  {
+    x->copyFromDev();
+    const local_ordinal_type N = getLocalSize(x);
+    auto* xdata = static_cast<const real_type*>(getLocalData(x));
+    
+    int local_fail = 0;
+
+    for(local_ordinal_type i = 0; i < N; ++i)
+    {
+      if(!isEqual(xdata[i], expect(i)))
+      {
+        ++local_fail;
+      }
+    }
+
+    return local_fail;
+  }
+
+  /// If test fails on any rank set fail flag on all ranks
+  bool reduceReturn(int failures, hiop::hiopVector* x)
+  {
+    int fail = 0;
+
+#ifdef HIOP_USE_MPI
+    if (auto* vecPar = dynamic_cast<hiop::hiopVectorPar*>(x))
+    {
+      MPI_Allreduce(&failures, &fail, 1, MPI_INT, MPI_SUM, vecPar->get_mpi_comm());
+    }
+    else if (auto* vecRajaPar = dynamic_cast<hiop::hiopVectorRajaPar*>(x))
+    {
+      MPI_Allreduce(&failures, &fail, 1, MPI_INT, MPI_SUM, vecRajaPar->get_mpi_comm());
+    }
+    else
+    {
+      //std::cout << "MPIComm not initialised for this vector implimentation.\n"; 
+      return false;
+    }
+#else
+    fail = failures;
+#endif
+    return (fail != 0);
+  }
+
+  /// Wrap new command
+  real_type* createLocalBuffer(local_ordinal_type N, real_type val)
```

Comment:
There might be a better way to do this, but this is the easiest solution that I could come up with that allows the test to be implementation agnostic. Also applies to `deleteLocalBuffer` below this function.

Reply:
This is also why I needed the additional include to umpire at the top of the file